### ISSUE TITLE
ValueError: Duplicate model found

### ISSUE THREAD

**Issue by nikitastupin** [Yelp/bravado#463]
Hi,

I've stumbled upon an error while parsing `https://firefox.settings.services.mozilla.com/v1/__api__` specification:

```
...
ValueError: Duplicate "SchemaNode" model found at "https://firefox.settings.services.mozilla.com/v1/__api__#/paths//__api__/get/responses/200/schema". Original "SchemaNode" model at "https://firefox.settings.services.mozilla.com/v1/__api__#/paths///get/responses/200/schema"
```

I expect it to be parsed OK as `https://inspector.swagger.io/builder` does.

# Versions

```
$ pip3 list | grep bravado
bravado                10.6.2
bravado-core           5.17.0 
```

# Steps to reproduce

```
python3 -c 'from bravado.client import SwaggerClient; SwaggerClient.from_url("https://firefox.settings.services.mozilla.com/v1/__api__")'
```

**Comment by nikitastupin** [comment]
As far as I understood we can fix this by not raising https://github.com/Yelp/bravado-core/blob/015a2cd8baf1ecdf0af2739452871ee19bc7b3d1/bravado_core/model.py#L71 exception. However I'm not sure side effects :smiley:

**Comment by macisamuele** [comment]
@nikitastupin Thanks a lot for reporting this. I've done some investigations, results are down the comment.

**TL;DR;** You could configure bravado to not use models in order to workaround the issue.

```
(venv) maci:~/ $ # It works
(venv) maci:~/ $ python3 -c 'from bravado.client import SwaggerClient; SwaggerClient.from_url("https://firefox.settings.services.mozilla.com/v1/__api__",config={"use_models": False})' &> /dev/null; echo $?

0
(venv) maci:~/ $ # It fails (as reported)
(venv) maci:~/ $ python3 -c 'from bravado.client import SwaggerClient; SwaggerClient.from_url("https://firefox.settings.services.mozilla.com/v1/__api__",config={"use_models": True})' &> /dev/null; echo $?

1
(venv) maci:~/ $                                                                                                                                                         [21:47:32]
```

---

**Details**
The reported issue is caused by the fact that `bravado`, using the default configurations, tries to define python classes that represent the models available on the specifications.
NOTE: The models discovery is actually performed by [`bravado-core`](https://github.com/Yelp/bravado-core) and documented in [here](https://bravado-core.readthedocs.io/en/stable/models.html).

The python classes are named according to the `x-model` attribute, if present, or `title` as fallback.
The specifications report multiple usages of `title: SchemaNode`, `title: ErrorSchema`, etc. (full list at the end of this comment).

In order to prevent failures you might configure the `bravado` client to not use models (in this case a warning will be rendered) (documented in [here](https://bravado.readthedocs.io/en/stable/configuration.html)).

A possible improvement (and I would suggest opening it into the bravado-core project) could be to make the model discovery smarter by allowing namesake models to coexist if their definition is exactly the same (as seems happening with the mozilla specifications).

Considering the above mentioned details this does not seems being an issue into bravado, but rather a potential feature request for bravado-core (which I would recommend opening in the correct project).

---

Full list of duplicated model names
```
(venv) maci:~/ $ curl --silent https://firefox.settings.services.mozilla.com/v1/__api__ | jq . | grep title | cut -d: -f2 | cut -d'"' -f2 | sort | uniq -dc | sort -n    [21:41:32]
      2 Cache Expires
      2 Collection
      2 Members
      2 Method
      3 Body
      3 Headers
      3 Path
      4 ObjectSchema
      4 Permissions
      4 Read
      4 Write
      7 Schema
      7 SchemaNode
     60 Code
     60 Errno
     60 Error
     60 ErrorSchema
     60 Info
     60 Message
(venv) maci:~/ $                                                                                                                                                         [21:41:36]
```
NOTE: The extraction is done in best-effort mode (aka there might be false-positive/false-negative) but is representative enough of the issue.
